### ISSUE TITLE
Add AppVeyor configuration

### ISSUE THREAD

**Issue by jimmysoda** [PDAL/CAPI#2]
Windows-based CI for is currently done on Simverge's private GitLab server.

* Create an AppVeyor configuration file (appveyor.yml) for building and testing pdal-c in Windows, using the existing GitLab CI (.gitlab-ci.yml) and PDAL's appveyor.yml as examples 
* Add an AppVeyor build badge to the top-level README.md

There is already an AppVeyor project for [Simverge/pdal-c](https://github.com/Simverge/pdal-c) under @jimmysoda's account. We can use this setup to make sure the AppVeyor configuration works as expected when complete.

**Comment by jimmysoda** [comment]
Here are a few options for addressing this issue:
* **Use vcpkg** - pdal-c was recently added to vcpkg but tests are not built; update vcpkg configuration to build the tests and call the test_pdalc.exe application in the AppVeyor configuration file
* **Use conda** - Follow the example from PDAL's [appveyor.yml](https://github.com/PDAL/PDAL/blob/master/appveyor.yml) and use conda; this approach is blocked by #3 
* **Build from source** - Clone repo with git, configure with CMake, build Visual Studio solution, and finally run test application

**Comment by runette** [comment]
I am assuming that when we get Conda working - this is no longer needed - see #6

**Comment by hobu** [comment]
I would suggest using github actions instead of appveyor. If contributors leave the config and the ability to manage it will stay with the the project.

**Comment by runette** [comment]
Just to close this off - I have created Build Tests for Windows / Linux and Mac as GH Actions. These run on PR and act as checks on the PR. I have added these badges in the README.

I also added a code style check as GH Action on PR.

We should now be totally running on GH Actions.

@jimmysoda  - there are probably some config files and links to your old environment that we don't need anymore.
